pkg/csi_driver/multishare_ops_manager.go

```diff
@@ -393,11 +388,26 @@ func (m *MultishareOpsManager) runEligibleInstanceCheck(ctx context.Context, req
 		}
 
 		klog.Infof("Instance %s/%s/%s with state %s is not ready with ongoing operation %s type %s", instance.Project, instance.Location, instance.Name, instance.State, op.Id, op.Type.String())
-		nonReadyInstanceCount += 1
+		nonReadyEligibleInstances = append(nonReadyEligibleInstances, instance)
+
 		// TODO: If we see > 1 instances with 0 shares (these could be possibly leaked instances where the driver hit timeout during creation op was in progress), should we trigger delete op for such instances? Possibly yes. Given that instance create/delete and share create/delete is serialized, maybe yes.
 	}
 
-	return readyEligibleInstances, nonReadyInstanceCount, nil
+	if len(readyEligibleInstances) == 0 && len(nonReadyEligibleInstances) > 0 {
+		errorString := "All eligible instances are busy.\n"
+
+		for _, instance := range nonReadyEligibleInstances {
+			op, _ := containsOpWithInstanceTargetPrefix(instance, ops) // Error for this call is already checked above
+			errorString = fmt.Sprintf("%s Instance %s busy with operation type %s\n", errorString, instance.Name, op.Type)
```

Comment:
for the scenario of line 360 above, where instance state != READY, I think its better report Instance is in "x" state, instead of ops.

For instance.state == READY, we can report busy with operation {opid}

Thoughts?

Reply:
Good point. Changed it to report state if OP isn't available.